### PR TITLE
doc/devguide: add page on Frame support - v4

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata/pull/6718

Describe changes:
- Remove parts of the explanation that would make more sense from the user's perspective, or maybe for a blog post on the new feature
- Reorganize contents, give more emphasis on how to add frame support to an applayer parser
- Update code snippets
- Remove note about being a work in progress

TODO:
- once the page about the frame keyword has been merged, update this one to link to it

Link to redmine ticket: 
- https://redmine.openinfosecfoundation.org/issues/4697

Compiled version can be seen at 
https://stream-frame-support-v4.surge.sh/extending/app-layer/app-layer-frames.html

### CODE REVIEW

**Comment by jufajardini** [comment]
Force pushed the removal of a change that should not be.

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    :start-at: pub fn new_tc
+    :lines: 1-6
+    :dedent: 4
+
+The parameters represent:
+
+- ``flow``: dedicated data type to carry specific flow-related data
+- ``stream_slice``: dedicated data type to carry stream data, shown bellow
+- ``frame_start``: a pointer to the start of the frame buffer in the stream, ``cur_i`` in the example code snippet
+- ``frame_len``: what we expect the frame length to be (the engine may need to wait until it has enough data...)
```
Comment (jufajardini) [inline doc/devguide/extending/app-layer/app-layer-frames.rst:89]: I think this may require a bit more explanation. Maybe telnet example?

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/6933

File: doc/devguide/extending/app-layer/app-layer-frames.rst (hunk 1 above)
Comment (victorjulien) [reply doc/devguide/extending/app-layer/app-layer-frames.rst:89]: "if the length is unknown pass -1"
"length can exceed the current input"

In telnet or other protocols that search for a end of frame char we'd generally start with length unknown, then update the length when the char is found

In smb and other binary protocols where the frames have explicit length fields we set frames to match those lengths, even if the length is bigger than the current input

**Pull request closed without merging**